web/pages/Chub/ChubItem.tsx

```diff
@@ -0,0 +1,114 @@
+import { Component, Show, createSignal } from 'solid-js'
+import Button from '/web/shared/Button'
+import { NewCharacter } from '/web/store'
+import { jsonToCharacter } from '../Character/ImportCharacter'
+import ChubImportCharModal from './ChubImportChar'
+import { CHUB_URL } from '/web/store/chub'
+import ChubImportBookModal from './ChubImportBook'
+import { extractCardData } from '../Character/card-utils'
+
+export const ChubItem: Component<{
+  name: string
+  fullPath: string
+  avatar: string
+  book?: boolean
+}> = (props) => {
+  const [importChar, setImportChar] = createSignal<NewCharacter>()
+  const [showCharModal, setShowCharModal] = createSignal<boolean>(false)
+  const [showBookModal, setShowBookModal] = createSignal<boolean>(false)
+
+  const [memorybook, setMemoryBook] = createSignal<any>({})
+
+  const processImage = async (file: File) => {
+    const json = await extractCardData(file)
+    if (!json) {
+      throw new Error('Invalid tavern image')
+    }
+    setImportChar(Object.assign(jsonToCharacter(json), { avatar: file }))
+    setShowCharModal(true)
+  }
+
+  const processItem = async () => {
+    const headers = {
+      'Content-Type': 'application/json',
+      accept: '/',
+    }
+
+    const body = {
+      format: 'tavern',
+      fullPath: props.fullPath,
+      version: 'main',
+    }
+
+    if (props.book) {
+      body.format = 'AGNAI'
+      const res = await fetch(`${CHUB_URL}/lorebooks/download`, {
+        headers: headers,
+        body: JSON.stringify(body),
+        method: 'post',
+      })
+
+      const json = await res.json()
+      setMemoryBook(json)
+    } else {
+      const res = await fetch(`${CHUB_URL}/characters/download`, {
+        headers: headers,
+        body: JSON.stringify(body),
+        method: 'post',
+      })
+
+      const blob = await res.blob()
+
+      processImage(new File([blob], `main_${props.fullPath}.png`))
```

Comment:
You'll need to set the mime type  here as the 3rd argument: `{ type: 'image/png' }` otherwise it'll be rejected by the Agnai api.